librz/core/cfile.c

```diff
@@ -697,6 +697,12 @@ static int rz_core_file_do_load_for_debug(RzCore *r, ut64 baseaddr, const char *
 #endif
 	}
 	rz_core_bin_apply_all_info(r, binfile);
+	if (r->dbg->cur->reg_profile) {
+		char *p = r->dbg->cur->reg_profile(r->dbg);
+		rz_reg_set_profile_string(r->dbg->reg, p);
+		rz_debug_reg_sync(r->dbg, RZ_REG_TYPE_ALL, 0);
+		free(p);
```

Comment:
shall this be done similar to what's done in https://github.com/rizinorg/rizin/pull/1359/commits/1bf66da0b25a9a586303f017df31002f45ec3738 ? Also, in that piece of code there is no `rz_debug_reg_sync`, is that ok? Maybe we should have a single function to perform this "set_profile_sync" operation, if there is not one already.